### DISCUSSION TITLE
Workaround for Importing document Object in Pyodide Running in a Web Worker

### DISCUSSION THREAD

**Discussion by rheanV** [pyodide/pyodide#5194] [Q&A] [answered]
I am running Pyodide in a web worker and would like to import the document object from the JavaScript module using the statement `from js import document`. However, this import fails probably because the document module cannot be found in the globalThis of the web worker, which does not have access to the DOM object. What would be a suitable workaround for this issue?

**Comment by ryanking13** [comment, marked as answer]
You need to communicate with the main thread to access the dom object. Pyodide itself does not provide any helper functions for that, but some downstream projects has their own solutions, so I think checking those project would be helpful: e.g. [PyScript Worker](https://docs.pyscript.net/2024.6.2/user-guide/workers/)

**Reply by rheanV** [reply]
Got it. Thanks a lot!
